OWML.ModHelper/ModInputHandler.cs

```diff
@@ -0,0 +1,447 @@
+﻿using System;
+using System.Reflection;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using OWML.Common;
+using UnityEngine;
+
+namespace OWML.ModHelper
+{
+	public class ModInputHandler:IModInputHandler
+	{
+		private class InputInterceptor
+		{
+			public static void SAxisRemPre(SingleAxisCommand __instance)
+			{
+				ModInputHandler._self.UnregisterGamesBinding(__instance);
+			}
+
+			public static void DAxisRemPre(DoubleAxisCommand __instance)
+			{
+				ModInputHandler._self.UnregisterGamesBinding(__instance);
+			}
+
+			public static void SAxisUpdPost(SingleAxisCommand __instance)
+			{
+				KeyCode pos, neg;
+				ModInputHandler._self.RegisterGamesBinding(__instance);
+				FieldInfo val = typeof(SingleAxisCommand).GetField("_value", BindingFlags.NonPublic | BindingFlags.Instance);
+				float curval = (float)(val.GetValue(__instance));
+				int axisdir = 1;
+				if (OWInput.UsingGamepad())
+				{
+					axisdir = (int)(typeof(SingleAxisCommand).GetField("_axisDirection", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(__instance));
```

Comment:
Maybe I'm missing something, but you have access to instance variables in the patch method parameters. You don't need to do all this, just add the parameter `int ____axisDirection` (3 extra underscores). Use ref if you need to write to it (no need for SetValue)

https://github.com/pardeike/Harmony/wiki/Patching#patch-parameters

Reply:
done (actually looks neater now, forgot about that function)